kiali/kiali
# Get version info for remote istiod and refactor status

### Describe the change

Gets the version for a remote istiod by port fowarding to the istiod pod and doing a `/version` request to the http monitoring port (15014).

Updates the default for `config.ExternalServices.Istio.UrlServiceVersion` to be `""`. If it is set to empty string, it will be autodetected and no longer needs to be set manually especially with multiple revisions. If it is set, the value will be used on the home cluster.

Also refactors some things out of `status/status` since there were circular imports with the `business` package.

### Steps to test the PR

1. Setup local multi-primary with kind
```
hack/run-integration-tests.sh --test-suite frontend-multi-primary --setup-only true
```
2. Edit an istiod deployment's image on the "west" cluster down to n-1 e.g. from `1.22.1` to `1.21.0`
3. Visit the mesh page and see the different version on the remote cluster: 
![Screenshot from 2024-06-10 11-00-12](https://github.com/kiali/kiali/assets/6226732/ca124fe9-7427-4728-92db-b236469a26c9)

### Automation testing

None created.

### Issue reference

Relates to #7177 and https://github.com/kiali/kiali/issues/7203
